### PR TITLE
Cuda backend2

### PR DESCRIPTION
**Pull request by cosunae**
Technical Description
===================
Add timers (used for cuda backend)

### CODE REVIEW

**Comment by cosunae** [comment]
launch jenkins

**Comment by cosunae** [comment]
launch jenkins

File: bundle/CMakeLists.txt [hunk 1]
```diff
@@ -90,8 +90,8 @@ yoda_find_package(
   REQUIRED_VARS dawn_DIR
   ADDITIONAL
     DOWNLOAD_DIR ${YODA_DOWNLOAD_DIR}
-    GIT_REPOSITORY "git@github.com:MeteoSwiss-APN/dawn.git"
-    GIT_TAG "master" 
+    GIT_REPOSITORY "git@github.com:cosunae/dawn.git"
```
Comment (twicki) [inline bundle/CMakeLists.txt]: change this back before merging
```suggestion
    GIT_REPOSITORY "git@github.com:MeteoSwiss-APN/dawn.git"
```

File: bundle/CMakeLists.txt [hunk 2]
```diff
@@ -90,8 +90,8 @@ yoda_find_package(
   REQUIRED_VARS dawn_DIR
   ADDITIONAL
     DOWNLOAD_DIR ${YODA_DOWNLOAD_DIR}
-    GIT_REPOSITORY "git@github.com:MeteoSwiss-APN/dawn.git"
-    GIT_TAG "master" 
+    GIT_REPOSITORY "git@github.com:cosunae/dawn.git"
+    GIT_TAG "cuda_backend2"
```
Comment (twicki) [inline bundle/CMakeLists.txt]:
```suggestion
    GIT_TAG "master" 
```

File: src/gridtools/clang/timer.hpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+  For information: http://eth-cscs.github.io/gridtools/
+*/
+#pragma once
```
Comment (twicki) [inline src/gridtools/clang/timer.hpp]: we've used ifdefs so far. I would stick to them

File: src/gridtools/clang/timer.hpp [hunk 4]
```diff
@@ -0,0 +1,106 @@
+/*
+  GridTools Libraries
```
Comment (twicki) [inline src/gridtools/clang/timer.hpp]: can we use GT-code like this, how does licensing work?

File: src/gridtools/clang/timer_cuda.hpp [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+  For information: http://eth-cscs.github.io/gridtools/
+*/
+#pragma once
```
Comment (twicki) [inline src/gridtools/clang/timer_cuda.hpp]: pragma vs ifndef

File: test/integration-test/CodeGen/conditional_stencil.cpp [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    }
+  }
+};
+
+stencil conditional_stencil {
+  storage in, out;
+
+  Do {
+    vertical_region(k_start, k_end) {
+      if(var2)
```
Comment (twicki) [inline test/integration-test/CodeGen/conditional_stencil.cpp]: can we have brackets here?

File: test/integration-test/CodeGen/conditional_stencil_benchmark.cpp [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+  storage_t in(meta_data, "in"), out_gt(meta_data, "out-gt"), out_naive(meta_data, "out-naive");
+
+  verif.fillMath(8.0, 2.0, 1.5, 1.5, 2.0, 4.0, in);
+  verif.fill(-1.0, out_gt, out_naive);
+
+  OPTBACKEND::conditional_stencil conditional_gt(dom, in, out_gt);
+  cxxnaive::conditional_stencil conditional_naive(dom, in, out_naive);
+
+  conditional_naive.set_var2(true);
+  conditional_gt.set_var2(true);
```
Comment (twicki) [inline test/integration-test/CodeGen/conditional_stencil_benchmark.cpp]: :+1:

File: src/gridtools/clang/timer.hpp (hunk 4 above)
Comment (twicki) [reply src/gridtools/clang/timer.hpp]: and if we have it as internal, we should use GTClang-Headers. If not, does it make sense to put it into an external folder?

File: src/gridtools/clang/timer.hpp (hunk 3 above)
Comment (cosunae) [reply src/gridtools/clang/timer.hpp]: done

File: test/integration-test/CodeGen/conditional_stencil.cpp (hunk 6 above)
Comment (cosunae) [reply test/integration-test/CodeGen/conditional_stencil.cpp]: done

**Comment by twicki** [comment]
launch pipeline

**Pull request merged**